server/routes/applications.js

```diff
@@ -66,20 +66,24 @@ router.get('/manage-applications/:appId/view/:env', (req, res, next) => {
   logger.info(`GET request to view a single application: ${req.path}`);
   const id = req.params.appId;
   const env = req.params.env;
-  const viewData = routeUtils.createViewData('View application', 'view-application');
+  const viewData = {
+    this_data: {
+      appId: req.params.appId,
+      env: env
+    },
+    this_errors: null,
+    active_page: 'view-application',
+    title: 'View application'
+  };
```

Comment:
Initially I switched to the helper function, but for some reason the appId stopped persisting onto the view page, and so I couldn't redirect to add key. I wasn't able to figure out why that was, so I just changed it back.....

Reply:
Try changing the `const viewData` to `let viewData`. 
If that doesn't work try
```
const viewData = routeUtils.createViewData('View application', 'view-application');
viewData.this_data.appId = req.params.appId;
viewData.this_data.env = env;

```
Or 
```
const viewData = routeUtils.createViewData('View application', 'view-application');
viewData.this_data = {
      appId: req.params.appId,
      env: env
};
```

Reply:
I've tried all suggested alternative information but I still have the same problem with appId failing to persist onto the next page. At this point I'm not sure how much more time we should spend on it, since the implementation I have still works

Reply:
Ok that's fine, you'll probably need to do a merge from master in your branch, i've added the authentication stuff which affects a lot of this. If you get any trouble with it give me a call.